pkg/kubelet/dockershim/network/cni/cni.go

```diff
@@ -280,14 +283,17 @@ func (plugin *cniNetworkPlugin) SetUpPod(namespace string, name string, id kubec
 		return fmt.Errorf("CNI failed to retrieve network namespace path: %v", err)
 	}
 
+	// Because the default remote runtime request timeout is 2 min
+	// Todo get the timeout from parent ctx
```

Comment:
Should it be slightly less than 2m so that we ensure the network timeout triggers before the remote runtime one does?

Reply:
The remote timeout is 2 times longer timeout for sandbox operation (4 mins by default).
And the RuntimeRequestTimeout is configurable.
So is hard to choose the cni timeout .
Now i choose 220s to slightly less than 4mins.
Maybe next time i can find a method to get the RuntimeRequestTimeout in grpc server.

Reply:
Should we just use the parent context? It seems more complicated to derive and set the timeout from the parent context?

Reply:
Created #82913 for this TODO

Reply:
> The remote timeout is 2 times longer timeout for sandbox operation (4 mins by default).
> And the RuntimeRequestTimeout is configurable.
> So is hard to choose the cni timeout .
> Now i choose 220s to slightly less than 4mins.
> Maybe next time i can find a method to get the RuntimeRequestTimeout in grpc server.

Can you tell me why the remote timeout is 2 times longer timeout for sandbox operation？ Thanks very much!